### ISSUE TITLE
Modin Dataframe Access is Slow

### ISSUE THREAD

**Issue by Ivan-Rodriguez-ck** [modin-project/modin#649]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04.6 LTS
- **Modin installed from (source or binary)**: binary
- **Modin version**: 0.5.0
- **Python version**: 3.6.5
- **Exact command to reproduce**:
```python
import random
from sklearn.cross_validation import train_test_split
import modin.pandas as pd


X = pd.DataFrame(np.random.randint(0, 100, size=(2000, 1)))
y = pd.Series(random.sample(range(1, 5000), 2000), index = random.sample(range(0, 5000), 2000))
X_train, X_test, y_train, y_test = train_test_split(X,y)

print(y_train.head())
```

### Describe the problem
This code is incredibly slow in modin compared to pandas, I wasnt able to get the issue to occur without running it through the sklearn train_test_split(). If the only change I make is the pd import between modin and pandas the pandas version completes in 0.001 seconds and the modin version takes 8 seconds.  Once the sklearn code runs any kind of access to either the modin dataframes or series that are returned by sklearn take inordinate amounts of time to complete.

**Comment by devin-petersohn** [comment]
Hi @Ivan-Rodriguez-ck, thanks for the question!

There are a couple of things you're running into here:

1.) Creating a `DataFrame`/`Series` from numpy will take time. The data has to be distributed: See #232. If you are using Modin, we recommend using one of the data ingest methods (`read_csv`, etc.)
2.) sklearn cannot use the distributed functionality of Modin yet, so it must be converted into a single threaded numpy array to be used.

If you have an interest in contributing to the project to work toward solutions for these limitations, let me know.

**Comment by aregm** [comment]
@devin-petersohn Will the latest work on train_test_split accelerate this? The difference is ~50x now.

**Comment by devin-petersohn** [comment]
Yes, this issue should be fixed now.

Closing this. Feel free to reopen if the discussion should continue or if issue was not resolved.

**Comment by HiIamJeff** [comment]
@devin-petersohn Hi Devin, I recently used Modin (0.7.3), NumPy (1.18.5), pandas (1.0.3) and scikit-learn (0.22.2.post1), and the issue is still there. I had a dataset with the shape of (369768, 58) (most of the columns are all float numbers), and the run time of Modin with train_test_split() is absurdly long (probably 30 minutes). I switched back to Pandas and the run time is less than 1 seconds. I think you might need to take a look at that and run some tests, or let me know if I made any mistake.

The only codes I switched between the tests
```
os.environ["MODIN_ENGINE"] = "dask"  # Modin will use Dask
import modin.pandas as pd
# import pandas as pd
```

The codes that took more than 10 minutes to run with Modin
```
X_train, X_test, y_train, y_test = train_test_split(df.drop([target_variable_name], axis=1), df[target_variable_name],
                                                        test_size=0.2, shuffle=True,
                                                        random_state = 111)
```

**Comment by devin-petersohn** [comment]
@HiIamJeff We are working on an experimental API for train_test_split because of the limitations in interacting with distributed objects from sklearn. The `shuffle` parameter of the `train_test_split` is causing all partitions to shuffle data between themselves, which is expensive. Link to PR: https://github.com/modin-project/modin/pull/1560
